### DISCUSSION TITLE
How to verify notation signed images in container build tools?

### DISCUSSION THREAD

**Discussion by cospeedster** [notaryproject/notation#984] [Q&A]
Hey together,

I'm wondering if someone else stubled over the "problem" to verify image signatures from base images during build with tools like Docker, Podman, Buildah and so on?

In my use case the build tool has to verify the container signature against provided policies or other trust anchors.

For exploration of this I used your [Quickstart Guide](https://notaryproject.dev/docs/quickstart-guides/quickstart-sign-image-artifact/). I was able to show and verify the signature with **notation**:

```bash
notation ls $IMAGE
localhost:5001/net-monitor@sha256:9c00e8f3ff2cf04bc47ceb80d83988fae1ee2db8d6905b03bc773e90654107b7
└── application/vnd.cncf.notary.signature
    └── sha256:1083e8ea5c0999a93c83bd6df7417fa0a1a088a2f8233f95ed80422a5880a308

notation verify $IMAGE
Successfully verified signature for localhost:5001/net-monitor@sha256:9c00e8f3ff2cf04bc47ceb80d83988fae1ee2db8d6905b03bc773e90654107b7
```

**Docker** seems not to be able to verify notation signed images with [Docker Content Trust](https://docs.docker.com/engine/security/trust/):

```bash
docker trust inspect --pretty $IMAGE
no signatures or cannot access localhost:5001/net-monitor@sha256:9c00e8f3ff2cf04bc47ceb80d83988fae1ee2db8d6905b03bc773e90654107b7
```

For **Podman** and **Buildah** it seems that they use the [_policy.json_](https://github.com/containers/image/blob/main/docs/containers-policy.json.5.md) which currently support only GPG keys to verify against ([source](https://github.com/containers/image/blob/main/docs/containers-policy.json.5.md#signedby))

Is there any chance to get this working or are there features planned to be compatible to these tools?

**Comment by yizha1** [comment]
@cospeedster Docker Content Trust cannot be used to verify notation signed images, as they are different solutions. May I know which pipeline you are using, as we have [notation GitHub actions](https://github.com/notaryproject/notation-action) which can be used for signing and verification.

Would you mind sharing any thoughts or work you may have related to the integration with these build tools? Are you willing to contribute to this work?
